chef/cookbooks/cpe_utils/libraries/node_functions.rb

```diff
@@ -284,6 +284,30 @@ def virtual?
       end
     end
 
+    # Granular virtual device type for macOS
+    def virtual_macos_type
+      unless macos?
+        Chef::Log.warn('node.virtual_macos called on non-OS X!')
+        return
+      end
+      return node['virtual_macos'] if node['virtual_macos']
+      if node['hardware']['boot_rom_version'].include? 'VMW'
+        virtual_type = 'vmware'
+      elsif node['hardware']['boot_rom_version'].include? 'VirtualBox'
+        virtual_type = 'virtualbox'
+      else
+        virtual_type = Mixlib::ShellOut.new(
```

Comment:
Use `shell_out` instead of Mixlib::Shellout.new. It is the supported way to do shellouts now.

Reply:
moved to shell_out now, but ironically after reading the code, they just do Mixlib:Shellout. 😄